### PR TITLE
Blacklist peer which didn't finish handshake in time

### PR DESCRIPTION
**Pull request by siy**

### CODE REVIEW

File: radixdlt-core/radixdlt/src/main/java/com/radixdlt/network/p2p/PendingOutboundChannelsManager.java [hunk 1]
```diff
@@ -133,18 +137,16 @@ private void handlePeerConnected(PeerEvent.PeerConnected peerConnected) {
...
+          .map(RadixNodeUri::getNodeId)
+          .flatMap(nodeId -> Optional.ofNullable(this.pendingChannels.remove(nodeId)))
           .ifPresent(
-              uri -> {
-                final var maybeFuture = this.pendingChannels.remove(uri.getNodeId());
-                if (maybeFuture != null) {
-                  maybeFuture.completeExceptionally(new IOException("Peer connection failed"));
-                }
-              });
+              maybeFuture ->
```
Comment (LukasGasior1) [inline radixdlt-core/radixdlt/src/main/java/com/radixdlt/network/p2p/PendingOutboundChannelsManager.java:148]: it's no longer `maybe`, the future is always there :)

File: radixdlt-core/radixdlt/src/main/java/com/radixdlt/network/p2p/transport/PeerChannel.java [hunk 2]
```diff
@@ -154,35 +154,36 @@ public PeerChannel(
...
-    this.addressing = Objects.requireNonNull(addressing);
-    this.peerEventDispatcher = Objects.requireNonNull(peerEventDispatcher);
-    this.uri = Objects.requireNonNull(uri);
-    uri.ifPresent(u -> this.remoteNodeId = u.getNodeId());
+    this.counters = requireNonNull(counters);
+    this.addressing = requireNonNull(addressing);
+    this.peerEventDispatcher = requireNonNull(peerEventDispatcher);
+    this.uri = requireNonNull(uri);
+
+    uri.map(RadixNodeUri::getNodeId).ifPresent(nodeId -> this.remoteNodeId = nodeId);
```
Comment (LukasGasior1) [inline radixdlt-core/radixdlt/src/main/java/com/radixdlt/network/p2p/transport/PeerChannel.java:162]: previous version was shorter ;)

File: radixdlt-core/radixdlt/src/main/java/com/radixdlt/network/p2p/transport/PeerChannel.java [hunk 3]
```diff
@@ -154,35 +154,36 @@ public PeerChannel(
...
-            },
-            BackpressureOverflowStrategy.DROP_LATEST);
+            config.channelBufferSize(), this::onOverflow, BackpressureOverflowStrategy.DROP_LATEST);
 
     if (this.nettyChannel.isActive()) {
       this.init();
     }
   }
 
+  private void onOverflow() {
```
Comment (LukasGasior1) [inline radixdlt-core/radixdlt/src/main/java/com/radixdlt/network/p2p/transport/PeerChannel.java:179]: how about: `onInboundMessageBufferOverflow`?

**Review by LukasGasior1 — APPROVED** [review]
minor comments; LGTM

File: radixdlt-core/radixdlt/src/main/java/com/radixdlt/network/p2p/PendingOutboundChannelsManager.java (hunk 1 above)
Comment (siy) [reply radixdlt-core/radixdlt/src/main/java/com/radixdlt/network/p2p/PendingOutboundChannelsManager.java:148]: Fixed

File: radixdlt-core/radixdlt/src/main/java/com/radixdlt/network/p2p/transport/PeerChannel.java (hunk 2 above)
Comment (siy) [reply radixdlt-core/radixdlt/src/main/java/com/radixdlt/network/p2p/transport/PeerChannel.java:162]: Yes, but this one more clear in the expression of intent.

File: radixdlt-core/radixdlt/src/main/java/com/radixdlt/network/p2p/transport/PeerChannel.java (hunk 3 above)
Comment (siy) [reply radixdlt-core/radixdlt/src/main/java/com/radixdlt/network/p2p/transport/PeerChannel.java:179]: Fixed

File: radixdlt-core/radixdlt/src/main/java/com/radixdlt/network/p2p/transport/PeerChannel.java (hunk 2 above)
Comment (LukasGasior1) [reply radixdlt-core/radixdlt/src/main/java/com/radixdlt/network/p2p/transport/PeerChannel.java:162]: I disagree, but I think it's a matter of taste. I'm okay with keeping it the new way

**Pull request merged**